### ISSUE TITLE
Need keyboard shortcuts for buttons in custom dialog

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#71430]
In Mac OS you can press cmd+letter on many native prompts to select the option that starts with that letter. For example, I can press cmd+D to select "Don't Save". Would be great for my muscle memory to have this in the custom dialog too. Otherwise I have to count tabs carefully to hit the right button, or grab my mouse.

Similarly, some people will probably want the pneumonics that the windows native prompt has (like alt+n for "Don't Save").

**Comment by pavkum** [comment]
can we have a simple confirmation dialog with quick-pick options similar to simple file selection option?

**Comment by svipas** [comment]
@sbatten @roblourens  with new macOS Big Sur native dialog is small and it’s slower than custom dialog style. What I miss is the ability to use keyboard shortcuts like CMD+D to “Don’t Save”. How this could be enabled? If it’s currently impossible maybe I could help with PR? I will need some info to begin with, but I’m willing to help 🙂

**Comment by roblourens** [comment]
I have gotten used to using the arrow keys to pick the Don't Save button, then pressing enter. The keyboard shortcut would be better but I haven't missed it that much.

**Comment by svipas** [comment]
Yeah I see I can use arrows to select “Don’t Save”, I believe it will take some time to rewrite muscle memory, but still it would be good to have keyboard shortcut for “Don’t Save” to match native dialogs. And it still faster instead of pressing arrows and watching if focus is on “Don’t Save” and then pressing Enter instead of just doing CMD+D. Anyways, I’m glad issue exists and if some help needed I could help 🙃

**Comment by stephancasas** [comment]
If the intention is to imitate native OS behavior, it's also worth considering that the <kbd>Return</kbd> or <kbd>Enter</kbd> keys in macOS always trigger the visually-stressed/default dialogue option.

In the current implementation of custom VSCode dialogues, these keys instead trigger whichever option is selected via tab or arrow keys.

**Comment by bpasero** [comment]
My understanding from reading some resources is that only `Cmd+D` is a valid keyboard shortcut in native dialogs on macOS to trigger the "No"/"Do not Save" option. I cannot find evidence that other dialog buttons can be triggered via keybindings other than `Enter` for the default one and `Escape` for the cancel one.

Is that other peoples understanding as well?

**Comment by svipas** [comment]
Thanks a lot @bpasero!
